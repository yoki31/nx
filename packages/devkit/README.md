<p style="text-align: center;">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://raw.githubusercontent.com/nrwl/nx/master/images/nx-dark.svg">
    <img alt="Nx - Smart Monorepos · Fast CI" src="https://raw.githubusercontent.com/nrwl/nx/master/images/nx-light.svg" width="100%">
  </picture>
</p>

{{links}}

<hr>

# Nx: Smart Monorepos · Fast CI

Nx is a build system with built-in tooling and advanced CI capabilities. It helps you maintain and scale monorepos, both locally and on CI.

This package contains a set of utilities for creating Nx plugins.

{{content}}
