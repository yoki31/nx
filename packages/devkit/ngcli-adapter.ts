/**
 * @category Ng CLI Adapter
 */
// eslint-disable-next-line @typescript-eslint/no-restricted-imports
export {
  wrapAngularDevkitSchematic,
  mockSchematicsForTesting,
  NxScopedHost,
} from 'nx/src/adapter/ngcli-adapter';
