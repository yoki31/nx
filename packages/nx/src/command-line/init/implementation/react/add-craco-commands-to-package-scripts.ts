import { readJsonFile, writeJsonFile } from '../../../../utils/fileutils';

export function addCracoCommandsToPackageScripts(
  appName: string,
  isStandalone: boolean
) {
  const packageJsonPath = isStandalone
    ? 'package.json'
    : `apps/${appName}/package.json`;
  const distPath = isStandalone
    ? `dist/${appName}`
    : `../../dist/apps/${appName}`;
  const packageJson = readJsonFile(packageJsonPath);
  packageJson.scripts = {
    ...packageJson.scripts,
    start: 'nx exec -- craco start',
    serve: 'npm start',
    build: `cross-env BUILD_PATH=${distPath} nx exec -- craco build`,
    test: 'nx exec -- craco test',
  };
  writeJsonFile(packageJsonPath, packageJson);
}
