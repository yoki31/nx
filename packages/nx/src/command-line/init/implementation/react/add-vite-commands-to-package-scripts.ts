import { readJsonFile, writeJsonFile } from '../../../../utils/fileutils';

export function addViteCommandsToPackageScripts(
  appName: string,
  isStandalone: boolean
) {
  const packageJsonPath = isStandalone
    ? 'package.json'
    : `apps/${appName}/package.json`;
  const packageJson = readJsonFile(packageJsonPath);
  packageJson.scripts = {
    ...packageJson.scripts,
    start: 'nx exec -- vite',
    serve: 'nx exec -- vite',
    build: `nx exec -- vite build`,
    test: 'nx exec -- vitest',
  };
  writeJsonFile(packageJsonPath, packageJson, { spaces: 2 });
}
