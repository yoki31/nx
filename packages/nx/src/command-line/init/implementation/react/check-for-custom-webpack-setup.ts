import { readJsonFile } from '../../../../utils/fileutils';

export function checkForCustomWebpackSetup() {
  const packageJson = readJsonFile('package.json');
  const combinedDeps = {
    ...packageJson.dependencies,
    ...packageJson.devDependencies,
  };
  ['react-app-rewired', '@craco/craco'].forEach((pkg) => {
    if (combinedDeps[pkg]) {
      console.log(
        `Skipping migration due to custom webpack setup. Found "${pkg}" usage. Use --force to continue anyway.`
      );
      process.exit(1);
    }
  });
}
