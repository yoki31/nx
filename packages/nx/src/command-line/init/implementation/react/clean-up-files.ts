import { removeSync } from 'fs-extra';
import { readJsonFile, writeJsonFile } from '../../../../utils/fileutils';

export function cleanUpFiles(appName: string, isStandalone: boolean) {
  // Delete targets from project since we delegate to npm scripts.
  const projectJsonPath = isStandalone
    ? 'project.json'
    : `apps/${appName}/project.json`;
  const json = readJsonFile(projectJsonPath);
  delete json.targets;
  if (isStandalone) {
    if (json.sourceRoot) {
      json.sourceRoot = json.sourceRoot.replace(`apps/${appName}/`, '');
    }
    if (json['$schema']) {
      json['$schema'] = json['$schema'].replace(
        '../../node_modules',
        'node_modules'
      );
    }
  }
  writeJsonFile(projectJsonPath, json);

  removeSync('temp-workspace');

  if (isStandalone) {
    removeSync('babel.config.json');
    removeSync('jest.preset.js');
    removeSync('jest.config.ts');
    removeSync('libs');
    removeSync('tools');
  }
}
