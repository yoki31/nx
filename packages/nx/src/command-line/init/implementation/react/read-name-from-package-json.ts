import { fileExists, readJsonFile } from '../../../../utils/fileutils';

export function readNameFromPackageJson(): string {
  let appName = 'webapp';
  if (fileExists('package.json')) {
    const json = readJsonFile('package.json');
    if (
      json['name'] &&
      json['name'].length &&
      json['name'].replace(/\s/g, '').length
    ) {
      appName = json['name'].replace(/\s/g, '');
    }
  }
  return appName;
}
