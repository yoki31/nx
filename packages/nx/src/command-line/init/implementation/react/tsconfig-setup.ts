import { join } from 'path';
import {
  fileExists,
  readJsonFile,
  writeJsonFile,
} from '../../../../utils/fileutils';

const defaultTsConfig = (relativePathToRoot: string) => ({
  extends: relativePathToRoot
    ? join(relativePathToRoot, 'tsconfig.base.json')
    : './tsconfig.base.json',
  compilerOptions: {
    jsx: 'react',
    allowJs: true,
    esModuleInterop: true,
    allowSyntheticDefaultImports: true,
  },
  files: [],
  include: [],
  references: [
    {
      path: './tsconfig.app.json',
    },
    {
      path: './tsconfig.spec.json',
    },
  ],
});

const defaultTsConfigApp = (relativePathToRoot: string) => ({
  extends: './tsconfig.json',
  compilerOptions: {
    outDir: join(relativePathToRoot, 'dist/out-tsc'),
    types: ['node'],
  },
  files: [
    join(relativePathToRoot, 'node_modules/@nx/react/typings/cssmodule.d.ts'),
    join(relativePathToRoot, 'node_modules/@nx/react/typings/image.d.ts'),
  ],
  exclude: ['**/*.spec.ts', '**/*.spec.tsx'],
  include: ['**/*.js', '**/*.jsx', '**/*.ts', '**/*.tsx'],
});

const defaultTsConfigSpec = (relativePathToRoot: string) => ({
  extends: './tsconfig.json',
  compilerOptions: {
    outDir: join(relativePathToRoot, 'dist/out-tsc'),
    module: 'commonjs',
    types: ['jest', 'node'],
  },
  include: [
    '**/*.spec.ts',
    '**/*.spec.tsx',
    '**/*.spec.js',
    '**/*.spec.jsx',
    '**/*.d.ts',
  ],
  files: [
    join(relativePathToRoot, 'node_modules/@nx/react/typings/cssmodule.d.ts'),
    join(relativePathToRoot, 'node_modules/@nx/react/typings/image.d.ts'),
  ],
});

export function setupTsConfig(appName: string, isStandalone: boolean) {
  const tsconfigPath = isStandalone
    ? 'tsconfig.json'
    : `apps/${appName}/tsconfig.json`;
  const tsconfigAppPath = isStandalone
    ? 'tsconfig.app.json'
    : `apps/${appName}/tsconfig.app.json`;
  const tsconfiSpecPath = isStandalone
    ? 'tsconfig.spec.json'
    : `apps/${appName}/tsconfig.spec.json`;
  const tsconfigBasePath = isStandalone
    ? './tsconfig.base.json'
    : '../../tsconfig.base.json';
  const relativePathToRoot = isStandalone ? '' : '../../';
  if (fileExists(tsconfigPath)) {
    const json = readJsonFile(tsconfigPath);
    json.extends = tsconfigBasePath;
    if (json.compilerOptions) {
      json.compilerOptions.jsx = 'react';
    } else {
      json.compilerOptions = {
        jsx: 'react',
        allowJs: true,
        esModuleInterop: true,
        allowSyntheticDefaultImports: true,
      };
    }
    writeJsonFile(tsconfigPath, json);
  } else {
    writeJsonFile(tsconfigPath, defaultTsConfig(relativePathToRoot));
  }

  if (fileExists(tsconfigAppPath)) {
    const json = readJsonFile(tsconfigAppPath);
    json.extends = './tsconfig.json';
    writeJsonFile(tsconfigAppPath, json);
  } else {
    writeJsonFile(tsconfigAppPath, defaultTsConfigApp(relativePathToRoot));
  }

  if (fileExists(tsconfiSpecPath)) {
    const json = readJsonFile(tsconfiSpecPath);
    json.extends = './tsconfig.json';
    writeJsonFile(tsconfiSpecPath, json);
  } else {
    writeJsonFile(tsconfiSpecPath, defaultTsConfigSpec(relativePathToRoot));
  }
}
