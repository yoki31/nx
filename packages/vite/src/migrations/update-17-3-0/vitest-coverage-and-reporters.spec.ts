import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { Tree, addProjectConfiguration } from '@nx/devkit';
import fixCoverageThreshold from './vitest-coverage-and-reporters';

describe('vitest-coverage-threshold migration', () => {
  let tree: Tree;

  beforeEach(() => {
    tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
  });

  it('should not do anything if no coverage', async () => {
    addProject1(tree, 'demo');
    await fixCoverageThreshold(tree);
    expect(tree.read('apps/demo/vite.config.ts', 'utf-8')).toMatchSnapshot();
  });

  it('should not do anything if coverage but no thresholds', async () => {
    addProject2(tree, 'demo2');
    await fixCoverageThreshold(tree);
    expect(tree.read('demo2/vite.config.ts', 'utf-8')).toMatchSnapshot();
  });

  it('should move thresholds in threshold object - has 2', async () => {
    addProject3(tree, 'demo3');
    await fixCoverageThreshold(tree);
    expect(tree.read('demo3/vite.config.ts', 'utf-8')).toMatchSnapshot();
  });

  it('should move thresholds in threshold object - has 3', async () => {
    addProject4(tree, 'demo4');
    await fixCoverageThreshold(tree);
    expect(tree.read('demo4/vite.config.ts', 'utf-8')).toMatchSnapshot();
  });
  it('should move thresholds in threshold object - has all', async () => {
    addProject5(tree, 'demo5');
    await fixCoverageThreshold(tree);
    expect(tree.read('demo5/vite.config.ts', 'utf-8')).toMatchSnapshot();
  });

  it('should move thresholds in threshold object - has 1', async () => {
    addProject6(tree, 'demo6');
    await fixCoverageThreshold(tree);
    expect(tree.read('demo6/vite.config.ts', 'utf-8')).toMatchSnapshot();
  });
});

function addProject1(tree: Tree, name: string) {
  addProjectConfiguration(tree, name, {
    root: `apps/${name}`,
    sourceRoot: `apps/${name}/src`,
    targets: {
      test: {
        executor: '@nx/vite:test',
        options: {},
      },
    },
  });

  tree.write(
    `apps/${name}/vite.config.ts`,
    `
/// <reference types="vitest" />
import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react';
import viteTsConfigPaths from 'vite-tsconfig-paths';

export default defineConfig({
  cacheDir: '../../node_modules/.vite/${name}',
  test: {
    globals: true,
    cache: {
      dir: '../../node_modules/.vitest',
    },
    environment: 'jsdom',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
  },
});

`
  );
}

function addProject2(tree: Tree, name: string) {
  addProjectConfiguration(tree, name, {
    root: `${name}`,
    sourceRoot: `${name}/src`,
    targets: {
      test: {
        executor: '@nx/vite:test',
        options: {},
      },
    },
  });

  tree.write(
    `${name}/vite.config.ts`,
    `
/// <reference types="vitest" />
import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react';
import viteTsConfigPaths from 'vite-tsconfig-paths';

export default defineConfig({
  cacheDir: '../../node_modules/.vite/${name}',
  test: {
    globals: true,
    cache: {
      dir: '../node_modules/.vitest',
    },
    environment: 'jsdom',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
    coverage: {
      reportsDirectory: '../coverage/${name}',
      provider: 'v8',
    },
  },
});
`
  );
}

function addProject3(tree: Tree, name: string) {
  addProjectConfiguration(tree, name, {
    root: `${name}`,
    sourceRoot: `${name}/src`,
    targets: {
      test: {
        executor: '@nx/vite:test',
        options: {},
      },
    },
  });

  tree.write(
    `${name}/vite.config.ts`,
    `
/// <reference types="vitest" />
import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react';
import viteTsConfigPaths from 'vite-tsconfig-paths';

export default defineConfig({
  cacheDir: '../../node_modules/.vite/${name}',
  test: {
    globals: true,
    cache: {
      dir: '../node_modules/.vitest',
    },
    environment: 'jsdom',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
    coverage: {
      reportsDirectory: '../coverage/${name}',
      provider: 'v8',
      lines: 100,
      statements: 100,
    },
  },
});
`
  );
}

function addProject4(tree: Tree, name: string) {
  addProjectConfiguration(tree, name, {
    root: `${name}`,
    sourceRoot: `${name}/src`,
    targets: {
      test: {
        executor: '@nx/vite:test',
        options: {},
      },
    },
  });

  tree.write(
    `${name}/vite.config.ts`,
    `
/// <reference types="vitest" />
import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react';
import viteTsConfigPaths from 'vite-tsconfig-paths';

export default defineConfig({
  cacheDir: '../../node_modules/.vite/${name}',
  test: {
    globals: true,
    cache: {
      dir: '../node_modules/.vitest',
    },
    environment: 'jsdom',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
    reporters: ['default'],
    coverage: {
      reportsDirectory: '../coverage/${name}',
      provider: 'v8',
      lines: 100,
      statements: 100,
      branches: 75,
    },
  },
});
`
  );
}

function addProject5(tree: Tree, name: string) {
  addProjectConfiguration(tree, name, {
    root: `${name}`,
    sourceRoot: `${name}/src`,
    targets: {
      test: {
        executor: '@nx/vite:test',
        options: {},
      },
    },
  });

  tree.write(
    `${name}/vite.config.ts`,
    `
/// <reference types="vitest" />
import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react';
import viteTsConfigPaths from 'vite-tsconfig-paths';

export default defineConfig({
  cacheDir: '../../node_modules/.vite/${name}',
  test: {
    globals: true,
    cache: {
      dir: '../node_modules/.vitest',
    },
    environment: 'jsdom',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
    coverage: {
      reportsDirectory: '../coverage/${name}',
      provider: 'v8',
      lines: 100,
      statements: 100,
      branches: 75,
      functions: 60,
    },
  },
});
`
  );
}
function addProject6(tree: Tree, name: string) {
  addProjectConfiguration(tree, name, {
    root: `${name}`,
    sourceRoot: `${name}/src`,
    targets: {
      test: {
        executor: '@nx/vite:test',
        options: {},
      },
    },
  });

  tree.write(
    `${name}/vite.config.ts`,
    `
/// <reference types="vitest" />
import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react';
import viteTsConfigPaths from 'vite-tsconfig-paths';

export default defineConfig({
  cacheDir: '../../node_modules/.vite/${name}',
  test: {
    globals: true,
    cache: {
      dir: '../node_modules/.vitest',
    },
    environment: 'jsdom',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
    coverage: {
      reportsDirectory: '../coverage/${name}',
      provider: 'v8',
      branches: 75,
    },
  },
});
`
  );
}
