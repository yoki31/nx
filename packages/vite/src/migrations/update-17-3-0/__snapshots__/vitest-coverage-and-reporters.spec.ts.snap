// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vitest-coverage-threshold migration should move thresholds in threshold object - has 1 1`] = `
"/// <reference types="vitest" />
import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react';
import viteTsConfigPaths from 'vite-tsconfig-paths';

export default defineConfig({
  cacheDir: '../../node_modules/.vite/demo6',
  test: {
    reporters: ['default'],
    globals: true,
    cache: {
      dir: '../node_modules/.vitest',
    },
    environment: 'jsdom',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
    coverage: {
      thresholds: {
        branches: 75,
      },
      reportsDirectory: '../coverage/demo6',
      provider: 'v8',
    },
  },
});
"
`;

exports[`vitest-coverage-threshold migration should move thresholds in threshold object - has 2 1`] = `
"/// <reference types="vitest" />
import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react';
import viteTsConfigPaths from 'vite-tsconfig-paths';

export default defineConfig({
  cacheDir: '../../node_modules/.vite/demo3',
  test: {
    reporters: ['default'],
    globals: true,
    cache: {
      dir: '../node_modules/.vitest',
    },
    environment: 'jsdom',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
    coverage: {
      thresholds: {
        lines: 100,
        statements: 100,
      },
      reportsDirectory: '../coverage/demo3',
      provider: 'v8',
    },
  },
});
"
`;

exports[`vitest-coverage-threshold migration should move thresholds in threshold object - has 3 1`] = `
"/// <reference types="vitest" />
import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react';
import viteTsConfigPaths from 'vite-tsconfig-paths';

export default defineConfig({
  cacheDir: '../../node_modules/.vite/demo4',
  test: {
    globals: true,
    cache: {
      dir: '../node_modules/.vitest',
    },
    environment: 'jsdom',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
    reporters: ['default'],
    coverage: {
      thresholds: {
        lines: 100,
        statements: 100,

        branches: 75,
      },
      reportsDirectory: '../coverage/demo4',
      provider: 'v8',
    },
  },
});
"
`;

exports[`vitest-coverage-threshold migration should move thresholds in threshold object - has all 1`] = `
"/// <reference types="vitest" />
import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react';
import viteTsConfigPaths from 'vite-tsconfig-paths';

export default defineConfig({
  cacheDir: '../../node_modules/.vite/demo5',
  test: {
    reporters: ['default'],
    globals: true,
    cache: {
      dir: '../node_modules/.vitest',
    },
    environment: 'jsdom',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
    coverage: {
      thresholds: {
        lines: 100,
        statements: 100,
        functions: 60,
        branches: 75,
      },
      reportsDirectory: '../coverage/demo5',
      provider: 'v8',
    },
  },
});
"
`;

exports[`vitest-coverage-threshold migration should not do anything if coverage but no thresholds 1`] = `
"/// <reference types="vitest" />
import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react';
import viteTsConfigPaths from 'vite-tsconfig-paths';

export default defineConfig({
  cacheDir: '../../node_modules/.vite/demo2',
  test: {
    reporters: ['default'],
    globals: true,
    cache: {
      dir: '../node_modules/.vitest',
    },
    environment: 'jsdom',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
    coverage: {
      reportsDirectory: '../coverage/demo2',
      provider: 'v8',
    },
  },
});
"
`;

exports[`vitest-coverage-threshold migration should not do anything if no coverage 1`] = `
"/// <reference types="vitest" />
import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react';
import viteTsConfigPaths from 'vite-tsconfig-paths';

export default defineConfig({
  cacheDir: '../../node_modules/.vite/demo',
  test: {
    reporters: ['default'],
    globals: true,
    cache: {
      dir: '../../node_modules/.vitest',
    },
    environment: 'jsdom',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
  },
});
"
`;
